Xamarin.Forms.Core/Layout.cs

```diff
@@ -319,69 +319,85 @@ internal static void LayoutChildIntoBoundingRegion(View child, Rectangle region,
 				{
-					return;
+					var v = LogicalChildrenInternal[index] as VisualElement;
+					if (v != null && v.IsVisible && (!v.IsPlatformEnabled || !v.IsNativeStateConsistent))
+						return;
 				}
-				if (trigger == InvalidationTrigger.HorizontalOptionsChanged || trigger == InvalidationTrigger.VerticalOptionsChanged)
+
+				var view = child as View;
+				if (view != null)
 				{
-					ComputeConstraintForView(view);
+					// we can ignore the request if we are either fully constrained or when the size request changes and we were already fully constrainted
+					if ((trigger == InvalidationTrigger.MeasureChanged && view.Constraint == LayoutConstraint.Fixed) ||
+						(trigger == InvalidationTrigger.SizeRequestChanged && view.ComputedConstraint == LayoutConstraint.Fixed))
+					{
+						return;
+					}
+					if (trigger == InvalidationTrigger.HorizontalOptionsChanged || trigger == InvalidationTrigger.VerticalOptionsChanged)
+					{
+						ComputeConstraintForView(view);
+					}
 				}
-			}
-
-			_allocatedFlag = false;
-			if (trigger == InvalidationTrigger.RendererReady)
-			{
-				InvalidateMeasureInternal(InvalidationTrigger.RendererReady);
-			}
-			else
-			{
-				InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
-			}
-
-			s_resolutionList.Add(new KeyValuePair<Layout, int>(this, GetElementDepth(this)));
-			if (!s_relayoutInProgress)
-			{
-				s_relayoutInProgress = true;
 
-				// Rather than recomputing the layout for each change as it happens, we accumulate them in 
-				// s_resolutionList and schedule a single layout update operation to handle them all at once.
-				// This avoids a lot of unnecessary layout operations if something is triggering many property
-				// changes at once (e.g., a BindingContext change)
+				_allocatedFlag = false;
+				if (trigger == InvalidationTrigger.RendererReady)
+				{
+					InvalidateMeasureInternal(InvalidationTrigger.RendererReady);
+				}
+				else
+				{
+					InvalidateMeasureInternal(InvalidationTrigger.MeasureChanged);
+				}
 
-				Device.BeginInvokeOnMainThread(() =>
+				s_resolutionList.Add(new KeyValuePair<Layout, int>(this, GetElementDepth(this)));
+				if (!s_relayoutInProgress)
 				{
-					// if thread safety mattered we would need to lock this and compareexchange above
-					IList<KeyValuePair<Layout, int>> copy = s_resolutionList;
-					s_resolutionList = new List<KeyValuePair<Layout, int>>();
-					s_relayoutInProgress = false;
+					s_relayoutInProgress = true;
 
-					foreach (KeyValuePair<Layout, int> kvp in copy.OrderBy(kvp => kvp.Value))
+					// Rather than recomputing the layout for each change as it happens, we accumulate them in 
+					// s_resolutionList and schedule a single layout update operation to handle them all at once.
+					// This avoids a lot of unnecessary layout operations if something is triggering many property
+					// changes at once (e.g., a BindingContext change)
+
+					if (Dispatcher != null && Dispatcher.IsInvokeRequired)
```

Comment:
This shouldn't be checking `IsInvokeRequired`. The use of BeginInvokeOnMainThread here is not because `OnChildMeasureInvalidated` might be called on a non-UI thread; it's because the layout operations need to be queued up after the current set of layout changes have been processed. 

So we _always_ need this to be queued up on the main thread. 

It's a hack which relies on a side-effect of how the various marshaling implementations work, and the hack will be going away in a future version, but for now it should just be 

```
if(Dispatcher != null) { Dispatcher.BeginInvokeOnMainThread(ResolveLayoutChanges) }
else { Device.BeginInvokeOnMainThread(ResolveLayoutChanges); }
```

Reply:
In that case, I believe that we do not need the lock that was included on method ResolveLayoutChanges, since the execution will be queued, right?

Reply:
> In that case, I believe that we do not need the lock that was included on method ResolveLayoutChanges, since the execution will be queued, right?

Correct - we don't need it, because that stuff will always be happening on the UI thread.

Reply:
Ok! Removed! :)